src/app/code/community/Shopgate/Cloudapi/Model/Api2/Wishlists/Items/Rest.php

```diff
@@ -56,4 +56,27 @@ protected function validateWishListItem($wishlistItem, $exception = true)
 
         return true;
     }
+
+    /**
+     * Sanitizer for incoming data
+     *
+     * @param Mage_Wishlist_Model_Wishlist $wishlist
+     *
+     * @throws Exception
+     * @return array
+     */
+    protected function getWishlistItemIds($wishlist)
+    {
+        $wishlistItemIds = $this->getRequest()->getParam('wishlistItemIds');
+
+        if (is_null($wishlistItemIds)) {
+            return $wishlist->getItemCollection()->getAllIds();
+        }
+
+        if ($wishlistItemIds === '') {
+            $this->_critical(self::RESOURCE_REQUEST_DATA_INVALID);
+        }
+
+        return explode(',', $wishlistItemIds);
```

Comment:
I suggest you further sanitize to make sure the wishlistItemId's are numbers to avoid a sql injection attack. Although, maybe Zend takes care of it already. Either way, we should do it just in case. Something like `$sanitized_array = array_map( 'intval', $ids );` will do.

Reply:
Can't hurt to validate that. done